.github/workflows/publish.yml

```diff
@@ -0,0 +1,56 @@
+# This workflow is triggered three ways:
+#
+# 1. When a tag is created, the workflow will upload the package to
```

Comment:
You need to update these docs as well then

Reply:
Ah, I see what's happened here. You had intended `workflow_dispatch` as a way to trigger test PyPI uploads, but when you coded it up you had `workflow_dispatch` doing the final version.

I actually forgot that `setuptools_scm` will do that whole "dirty commit" signature if you do a build without a tag, which certainly makes your proposed workflow more attractive.

I suppose I'm fine with either way, whichever works for you.

Reply:
That is indeed exactly what I intended. Mirroring the workflow in tzdata but with an additional way to test the workflow.

Reply:
I'm also OK with this workflow, but we will need to remember triggering the releases manually (maybe that is actually a good thing)